src/tests/ftest/util/general_utils.py

```diff
@@ -134,6 +134,144 @@ def _calculate_metrics(cls, data):
         return max_time, min_time, avg_time
 
 
+class RemoteCommandResult():
+    """Stores the command result from a Task object."""
+
+    class ResultData():
+        # pylint: disable=too-few-public-methods
+        """Command result data for the set of hosts."""
+
+        def __init__(self, command, returncode, hosts, stdout, timeout):
+            """Initialize a ResultData object.
+
+            Args:
+                command (str): the executed command
+                returncode (int): the return code of the executed command
+                hosts (NodeSet): the host(s) on which the executed command yielded this result
+                stdout (list): the result of the executed command split by newlines
+                timeout (bool): indicator for a command timeout
+            """
+            self.command = command
+            self.returncode = returncode
+            self.hosts = hosts
+            self.stdout = stdout
+            self.timeout = timeout
+
+    def __init__(self, command, task):
+        """Create a RemoteCommandResult object.
+
+        Args:
+            command (str): command executed
+            task (Task): object containing the results from an executed clush command
+        """
+        self.output = []
+        self._process_task(task, command)
+
+    @property
+    def homogeneous(self):
+        """Did all the hosts produce the same output.
+
+        Returns:
+            bool: if all the hosts produced the same output
+
+        """
+        return len(self.output) == 1
+
+    @property
+    def passed(self):
+        """Did the command pass on all the hosts.
+
+        Returns:
+            bool: if the command was successful on each host
+
+        """
+        non_zero = any(data.returncode != 0 for data in self.output)
+        return not non_zero and not self.timeout
+
+    @property
+    def timeout(self):
+        """Did the command timeout on any hosts.
+
+        Returns:
+            bool: True if the command timed out on at least one set of hosts; False otherwise
+
+        """
+        return all(data.timeout for data in self.output)
+
+    def _process_task(self, task, command):
+        """Populate the output list and determine the passed result for the specified task.
+
+        Args:
+            task (Task): a ClusterShell.Task.Task object for the executed command
+            command (str): the executed command
+        """
+        # Get a dictionary of host list values for each unique return code key
+        results = dict(task.iter_retcodes())
+
+        # Get a list of any hosts that timed out
+        timed_out = [str(hosts) for hosts in task.iter_keys_timeout()]
```

Comment:
FYI, if you just want to convert the type of each element
```suggestion
        timed_out = list(map(str, task.iter_keys_timeout()))
```